lib/msf/core/post/linux/compile.rb

```diff
@@ -11,35 +11,43 @@ def initialize(info = {})
     super
     register_options( [
       OptEnum.new('COMPILE', [true, 'Compile on target', 'Auto', ['Auto', 'True', 'False']]),
+      OptEnum.new('COMPILER', [true, 'Compiler to use on target', 'gcc', ['gcc', 'clang']]),
     ], self.class)
   end
 
   def live_compile?
     return false unless %w{ Auto True }.include?(datastore['COMPILE'])
 
-    if has_gcc?
-      vprint_good 'gcc is installed'
-      return true
+    if datastore['COMPILER'] == gcc
+      if has_gcc?
+        vprint_good 'gcc is installed'
+        return true
+      end
+    elsif datastore['COMPILER'] == clang
+      if has_clang?
+        vprint_good 'clang is installed'
+        return true
+      end
```

Comment:
I don't see where `gcc` or `clang` are defined. It seems like they should be strings.

```suggestion
    if datastore['COMPILER'] == 'gcc' && has_gcc?
      vprint_good 'gcc is installed'
      return true
    elsif datastore['COMPILER'] == 'clang' && has_clang?
      vprint_good 'clang is installed'
      return true
    end
```